polus-outlier-removal-plugin/src/main.py

```diff
@@ -0,0 +1,132 @@
+    parser.add_argument('--methods', dest='methods', type=str,
+                        help='Select methods for outlier detection', required=True)
+    parser.add_argument('--types', dest='types', type=str,
+                        help='Select type of outliers to detect', required=False)
+    parser.add_argument('--outdir', dest='outdir', type=str,
+                        help='Output collection', required=True)
+    
+    # Parse the arguments
+    args = parser.parse_args()  
+    
+    #Path to csvfile directory
+    inpdir = args.inpdir
+    logger.info('inpdir = {}'.format(inpdir))
+    
+    #Detect outliers using different methods
+    methods = args.methods
+    logger.info('methods = {}'.format(methods))
+    
+    #Detect outliers based on type selected
+    types = args.types
+    logger.info('types = {}'.format(types))
+    
+    #Path to save output csvfiles
+    outdir = args.outdir
+    logger.info('outdir = {}'.format(outdir))
+    
+    #Get list of .csv files in the directory including sub folders for outlier removal
+    inputcsv = list_file(inpdir)
+    if not inputcsv:
+        raise ValueError('No .csv files found.')
+        
+    if not methods:
+       raise ValueError('Select methods for outlier detection.')
+            
+    #Dictionary of methods to detect outliers
+    FEAT = {'IsolationForest': isolationforest}
+       
+    for inpfile in inputcsv:
+        #Get the full path
+        split_file = os.path.normpath(inpfile)
+        
+        #split to get only the filename
+        inpfilename = os.path.split(split_file)
+        file_name_csv = inpfilename[-1]
+        file_name,file_name1 = file_name_csv.split('.', 1)
+        
+        #Read csv file
+        logger.info('Started reading the file ' + file_name)
+        df = pd.read_csv(inpfile)
+        
+        #Standardize the data
+        data = StandardScaler().fit_transform(df)
+        
+        #Detect outliers
+        logger.info('Detecting outliers using ' + methods)
+        rem_out = FEAT[methods](data,types)
+        df['anomaly']= rem_out
+        if types == 'Global':
+            inliers = df.loc[df['anomaly']==1]
+            outliers = df.loc[df['anomaly']==-1]
+        else:
+            inliers = df.loc[df['anomaly']==0]
+            outliers = df.loc[df['anomaly']==1]
+        
+        #Drop 'anomaly' column    
+        inliers = inliers.drop('anomaly',axis=1)
+        outliers = outliers.drop('anomaly',axis=1)
+
+    	#Save dataframe into csv file
+        os.chdir(outdir)
+        logger.info('Saving csv file')
+        export_inlier = inliers.to_csv (r'%s_inliers.csv'%file_name, index=None, header=True, encoding='utf-8-sig')
+        export_outlier = outliers.to_csv (r'%s_outliers.csv'%file_name, index=None, header=True, encoding='utf-8-sig')
+        logger.info("Finished all processes!")
+
+if __name__ == "__main__":
+    main()
```

Comment:
Consider doing argument parsing in the `if __name__ == "__main__":` block, then pass all parse arguments to `main()`. Also, use the `typing` module when creating the `main()` function to define inputs and outputs.

It is not necessary to implement anything in this comment, but in the future it may become a requirement. This will help us perform other useful actions on this plugin in the future.